src/gsi/crtm_interface.f90

```diff
@@ -2597,6 +2693,185 @@ subroutine calc_gfdl_reff(rho_air,tsen,qxmr,cloud_name,reff)
+  real(r_kind):: lam_i,lam_w,lam_s,lam_r, lam_g,lam_exp, am_r,am_w,am_i,am_g, N_0r ,N_0i,N_0w,mu_w
+
+  ! Parameters
+  real(r_kind), parameter :: qmin = 1.0e-12_r_kind          ! [kg/kg ]
+  !.. droplet number concentration.
+  real, parameter :: Nt_c = 100.E6  ![m-3]
+
+  ! Parameters for water cloud
+  real(r_kind), parameter :: ccn        =  1.0e8_r_kind
+  real(r_kind), parameter :: rho_w      = 1000.0_r_kind     ! [kg/m3 ]
+  real(r_kind), parameter :: reff_w_min =    2.0_r_kind     ! previous value was 5.0_r_kind
+  real(r_kind), parameter :: reff_w_max =   25.0_r_kind     ! previous value was 10.0_r_kind
+
+  ! Parameters for ice cloud (Hemisfield and mcFarquhar 1996)
+  real(r_kind), parameter :: rho_i      = 890.0_r_kind      ! [kg/m3 ]
+  real(r_kind), parameter :: reff_i_min =    2.5_r_kind     ! previous value was 10_r_kind
+  real(r_kind), parameter :: reff_i_max = 250.0_r_kind      ! previous value was 150_r_kind
+  real(r_kind), parameter:: mu_i = 0.0
+
+  ! Parameters for  rain (Lin 1983)
+  real(r_kind), parameter :: rho_r      =    1000.0_r_kind  ! [kg/m3 ]
+  real(r_kind), parameter :: reff_r_min =       50.0_r_kind  ! [micron] ! previous value was 0.0_r_kind
+  real(r_kind), parameter :: reff_r_max =   1000.0_r_kind  ! [micron] ! previous value was 10000.0_r_kind
+  real(r_kind), parameter:: mu_r = 0.0
+  ! Parameters for snow
+  real(r_kind), parameter :: reff_s_min =       5.0_r_kind  ! [micron] ! previous value was 0.0_r_kind
+  real(r_kind), parameter :: reff_s_max =   5000.0_r_kind  ! [micron] ! previous value was 10000.0_r_kind
+
+!For snow moments conversions  (from Field et al. 2005)
+  real, dimension(10), parameter:: &
+      sa = (/ 5.065339, -0.062659, -3.032362, 0.029469, -0.000285,      &
+     &        0.31255,   0.000204,  0.003199, 0.0,      -0.015952/)
+      real, dimension(10), parameter:: &
+      sb = (/ 0.476221, -0.015896,  0.165977, 0.007468, -0.000141,      &
+     &        0.060366,  0.000079,  0.000594, 0.0,      -0.003577/)
+  real(r_kind), parameter :: am_s      =     0.069_r_kind
+  real(r_kind), parameter :: bm_s = 2.0_r_kind
+  real, dimension(1), parameter :: cse = (/ bm_s + 1. /)
+  real :: tc0, smo2, smob, smoc, smoz, a_, b_, loga_
+
+  ! Parameters for graupel  (Lin 1983)
+  real(r_kind), parameter :: rho_g      =     500.0_r_kind  ! [kg/m3 ]
+  real(r_kind), parameter :: reff_g_min =       150.0_r_kind  ! [micron] ! previous value was 0.0_r_kind
+  real(r_kind), parameter :: reff_g_max =   5000.0_r_kind  ! [micron] ! previous value was 10000.0_r_kind
+  real(r_kind), parameter:: mu_g = 0.0
+  real(r_kind), parameter :: no_exp     =  1.0e-4_r_kind
+
+!cloud water
+  if (trim(cloud_name)=='ql') then
+     am_w = rho_w*pi/6.0
+     reff_min = reff_w_min
+     reff_max = reff_w_max
+     do k = 1, nsig
+        qx = qxmr(k) * rho_air(k)  ! convert mixing ratio (kg/kg) to water content (kg/m3)
+        if (qx < qmin) then
+               mu_w = MAX(2, MIN((NINT(1000.E6/Nt_c) + 2), 15))
```

Comment:
`mu_w` is declared with precision `r_kind`.  Here we assign `mu_w` with an integer value between 2 and 15.  Should `mu_w` be declared as an `integer(i_kind)`?